inc/saiuburst.h

```diff
@@ -0,0 +1,400 @@
+     * @objects SAI_OBJECT_TYPE_TAM
+     */
+    SAI_TAM_MICROBURST_ATTR_TAM_ID = SAI_TAM_MICROBURST_ATTR_START,
+
+    /**
+     * @brief Statistic for this microburst
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM_STAT
+     */
+    SAI_TAM_MICROBURST_ATTR_STATISTIC,
+
+    /**
+     * @brief Watermark Levels
+     *
+     * Breach high/low watermark level for this microburst statistic
+     * in number of bytes.
+     *
+     * If not specified, the microburst is created without any
+     * levels, which is effectively disabling the microburst
+     * monitoring for the statistic.
+     *
+     * @type sai_uint64_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_TAM_MICROBURST_ATTR_LEVEL_A,
+
+    /**
+     * @brief Watermark Levels
+     *
+     * Breach low/high watermark level for this microburst statistic
+     * in number of bytes.
+     *
+     * If not specified, the microburst is created without any
+     * levels, which is effectively disabling the microburst
+     * monitoring for the statistic.
+     *
+     * @type sai_uint64_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_TAM_MICROBURST_ATTR_LEVEL_B,
+
+    /**
+     * @brief Transporter Object
+     *
+     * Provides the transporter object for this microburst.
+     *
+     * In the absence of a transporter, the default transporter will
+     * be used.
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM_TRANSPORTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_TAM_MICROBURST_ATTR_TRANSPORTER,
+
+    /**
+     * @brief Statistics for inclusion in the microburst
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM_STAT
+     * @default empty
+     */
+    SAI_TAM_MICROBURST_ATTR_STATS,
+
+    /**
+     * @brief End of Attributes
+     */
+    SAI_TAM_MICROBURST_ATTR_END,
+
+    /** Custom range base value */
+    SAI_TAM_MICROBURST_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_TAM_MICROBURST_ATTR_CUSTOM_RANGE_END
+
+} sai_tam_microburst_attr_t;
+
+/**
+ * @brief TAM Histogram Attributes.
+ */
+typedef enum _sai_tam_histogram_attr_t
+{
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_TAM_HISTOGRAM_ATTR_START,
+
+    /**
+     * @brief TAM Object
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM
+     */
+    SAI_TAM_HISTOGRAM_ATTR_TAM_ID = SAI_TAM_HISTOGRAM_ATTR_START,
+
+    /**
+     * @brief Buffers/Statistics for inclusion in histogram
+     *
+     * Specifies the Statistics/Types for a histogram.
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM_MICROBURST
+     * @default empty
+     */
+    SAI_TAM_HISTOGRAM_ATTR_STAT_TYPE,
```

Comment:
"type" would suggest that this attribute should be enum and not object_list, can you explain why is that ?